ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/AgingImages.tsx

```diff
@@ -151,7 +161,14 @@ function getViewAllLink(searchFilter: SearchFilter) {
 function AgingImages() {
     const { isOpen: isOptionsOpen, onToggle: toggleOptionsOpen } = useSelectToggle();
     const { searchFilter } = useURLSearch();
-    const [timeRanges, dispatch] = useReducer(timeRangeReducer, defaultTimeRanges);
+    const { pathname } = useLocation();
+
+    const [{ timeRanges }, dispatch] = useWidgetConfig<Config, TimeRangeAction>(
```

Comment:
Why `dispatch` here, and `updateConfig` without any async/await in all the other widgets?

Reply:
The `useWidgetConfig` hook's second return value is a function to update the config state, which defaults to a simple merge function. If you pass a reducer function as the hook's 4th argument though, it will be used as the state update function instead of the default merge function.

I kept this as `dispatch` to be the same as when this component used `useReducer`, but I don't love the inconsistency either. Do you think calling it `updateConfig` everywhere is better, even if the usage is slightly different?

The async/await is because this is the one case where the update call is not the last line of a function (in the other cases I just `returned` the function call.) This is to get around the eslint `no-dangling-promises` error. The other alternative to the async/await is to `return` the function call as well, but with an extra `return undefined` at the end. I can update to do that an add a short comment for consistency.

Reply:
Ah, I didn't check the code inside `useWidgetConfig` because it was merged in an earlier PR.

I agree there is no really good solution. You could merge as-is; no need to belabor the point.

We should have a lunch-and-learn once all the team is back from PTO, and get feedback on the pattern. We may get some suggestions for future optimization.